Source/Private/Invoke-ExecCommandCore.ps1

```diff
@@ -0,0 +1,74 @@
+#   Borrowed from our friendly provider: https://github.com/dotnet/roslyn/blob/master/src/Setup/Installer/tools/utils.ps1
+
+function Invoke-ExecCommandCore {
+    param (
+        [Parameter(mandatory = $true)]
+        [string] $Command,
+        [string] $CommandArgs,
+        [switch] $PassThru = $true
+    )
+
+    $result = [CommandCoreResult]::new()
+    $result.Command = $Command
+    $result.CommandArgs = $CommandArgs
+
+    $startInfo = New-Object System.Diagnostics.ProcessStartInfo
+    $startInfo.FileName = $Command
+    $startInfo.Arguments = $CommandArgs
+
+    $startInfo.UseShellExecute = $false
+    $startInfo.WorkingDirectory = Get-Location
+
+    #   Always redirect output
+    $startInfo.RedirectStandardOutput = $true
+    $startInfo.RedirectStandardError = $true
+    $startInfo.CreateNoWindow = $true
+
+    $process = New-Object System.Diagnostics.Process
+    $process.StartInfo = $startInfo
+
+    try {
+        $process.Start() | Out-Null
```

Comment:
I only selected the first line by this comment is for the `try` block. I think this function has taken on too much responsibility. We should find a way to split the responsibility into separate functions so that this function consistently operates on the same level of abstraction.

Reply:
Yeah. The complexity here also handles `Ctrl+C` to break / abort long running processes, which i am still figuring out how to test.